7_python_flask_fastapi/lesson1/app_fastapi.py

```diff
@@ -0,0 +1,217 @@
+import logging
+import databases
+import sqlalchemy
+from fastapi import FastAPI
+from typing import Optional, List
+from pydantic import BaseModel, Field
+from datetime import datetime
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+DATABASE_URL = "sqlite:///mydatabase.db"
+database = databases.Database(DATABASE_URL)
+metadata = sqlalchemy.MetaData()
+
+users = sqlalchemy.Table(
+    "users",
+    metadata,
+    sqlalchemy.Column("id", sqlalchemy.Integer, primary_key=True),
+    sqlalchemy.Column("first_name", sqlalchemy.String(32)),
+    sqlalchemy.Column("last_name", sqlalchemy.String(32)),
+    sqlalchemy.Column("email", sqlalchemy.String(128)),
+    sqlalchemy.Column("password", sqlalchemy.String(128)),
+)
+
+products = sqlalchemy.Table(
+    "products",
+    metadata,
+    sqlalchemy.Column("id", sqlalchemy.Integer, primary_key=True),
+    sqlalchemy.Column("name", sqlalchemy.String(64)),
+    sqlalchemy.Column("description", sqlalchemy.String(256)),
+    sqlalchemy.Column("price", sqlalchemy.Float),
+)
+
+orders = sqlalchemy.Table(
+    "orders",
+    metadata,
+    sqlalchemy.Column("id", sqlalchemy.Integer, primary_key=True),
+    sqlalchemy.Column("user_id", sqlalchemy.ForeignKey("users.id")),
+    sqlalchemy.Column("product_id", sqlalchemy.ForeignKey("products.id")),
+    sqlalchemy.Column("order_date", sqlalchemy.DateTime),
+    sqlalchemy.Column("order_status", sqlalchemy.String(32)),
+)
+
+engine = sqlalchemy.create_engine(DATABASE_URL)
+metadata.create_all(engine)
+
+app = FastAPI()
+
+class UserIn(BaseModel):
+    first_name: str = Field(max_length=32)
+    last_name: str = Field(max_length=32)
+    email: str = Field(max_length=128)
+    password: str = Field(max_length=128)
+
+class User(BaseModel):
+    id: int
+    first_name: str = Field(max_length=32)
+    last_name: str = Field(max_length=32)
+    email: str = Field(max_length=128)
+
+class ProductIn(BaseModel):
+    name: str = Field(max_length=64)
+    description: str = Field(max_length=256)
+    price: float
+
+class Product(BaseModel):
+    id: int
+    name: str = Field(max_length=64)
+    description: str = Field(max_length=256)
+    price: float
+
+class orderIn(BaseModel):
+    user_id: int
+    product_id: int
+    order_date: Optional[datetime] = None
+    order_status: Optional[str] = None
+
+class order(BaseModel):
+    id: int
+    user_id: int
+    product_id: int
+    order_date: Optional[datetime] = None
+    order_status: Optional[str] = None
+
+@app.post("/users/", response_model=User)
+async def create_user(user: UserIn):
+    query = users.insert().values(
+        first_name=user.first_name,
+        last_name=user.last_name,
+        email=user.email,
+        password=user.password,
+    )
+    last_record_id = await database.execute(query)
+    return {**user.dict(), "id": last_record_id}
```

Comment:
Метод `dict` устарел, вместо него используйте метод `model_dump`.
```suggestion
    return {**user.model_dump(), "id": last_record_id}
```
или можно использовать функцию `vars`.
```suggestion
    return {**vars(user), "id": last_record_id}
```